returnn/torch/engine.py

```diff
@@ -519,9 +519,16 @@ def eval_model(self, *, skip_already_evaluated: bool = False):
 
             if self._torch_distributed_ctx and self._torch_distributed_ctx.rank() != 0:
                 # We need to make sure the data loader iterator was created for proper synchronization.
-                # However, now we only want to do evaluation on rank 0 for simplicity.
+                # However, we only want to do evaluation on rank 0 for simplicity.
                 iter(data_loader)
+                # We wait here until rank 0 is done w/ the eval.
+                # Regularly synchronizing should fix potential timeout issues, like
+                # https://github.com/rwth-i6/returnn/issues/1621.
+                _has_data = torch.tensor([True], device="cpu", dtype=torch.int8)
+                while _has_data:
+                    _has_data = torch.distributed.broadcast(_has_data, src=0)
```

Comment:
Ah there is a bug here: You should not assign `_has_data = ...`.

Anyway, see https://github.com/rwth-i6/returnn/issues/1621#issuecomment-2361048191.

Reply:
Oh, I see. I'll re-do this PR w/ the change.